### PR TITLE
Fix issue with credentials in terraform script

### PR DESCRIPTION
**Pull request by smahmed776**

### CODE REVIEW

File: src/env.js [hunk 1]
```diff
@@ -29,8 +24,8 @@ export const env = createEnv({
     GOOGLE_CLIENT_SECRET: z.string(),
     GOOGLE_REDIRECT_URI: z.string().url(),
 
-    AWS_ACCESS_KEY_ID: z.string(),
-    AWS_SECRET_ACCESS_KEY: z.string(),
+    AWS_ACCESS_KEY_ID: z.string().optional(),
```
Comment (itsjoeoui) [inline src/env.js:27]: could u elaborate on why making this optional?

File: src/lib/email.ts [hunk 2]
```diff
@@ -62,12 +62,14 @@ const sendEmailWithSES = async ({
     },
   };
   const ses = new SES({
-    region: "us-east-2",
-    credentials: {
-      accessKeyId: env.AWS_ACCESS_KEY_ID,
-      secretAccessKey: env.AWS_SECRET_ACCESS_KEY,
-    },
+    region: env.AWS_REGION,
+    credentials:
```
Comment (itsjoeoui) [inline src/lib/email.ts:66]: could you elaborate a bit more here as well? Thanks

**Review by itsjoeoui — CHANGES_REQUESTED** [review]

File: src/env.js (hunk 1 above)
Comment (smahmed776) [reply src/env.js:27]: AWS provides temporary credentials within the EC2 instance for accessing AWS resources when an IAM role with appropriate policy is attached to the instance. The AWS SDK automatically retrieves these credentials from the instance, eliminating the need for using long-lived access/secret keys.. [See](https://aws.amazon.com/blogs/security/a-safer-way-to-distribute-aws-credentials-to-ec2/)

File: src/lib/email.ts (hunk 2 above)
Comment (smahmed776) [reply src/lib/email.ts:66]: @itsjoeoui Credentials are automatically retrieved from EC2 instance when they are not provided explicitly.

**Review by itsjoeoui — APPROVED** [review]
lgtm

**Pull request merged**